// @polkadot/ui-settings/cjs/defaults/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.STORAGE = exports.STORAGE_DEFAULT = exports.METADATA_UP = exports.METADATA_UP_DEFAULT = exports.LOCKING = exports.LOCKING_DEFAULT = exports.LANGUAGE_DEFAULT = exports.CAMERA = exports.CAMERA_DEFAULT = exports.UITHEMES = exports.UITHEME_DEFAULT = exports.UIMODES = exports.UIMODE_DEFAULT = exports.NOTIFICATION_DEFAULT = exports.ICONS = exports.ICON_DEFAULT_HOST = exports.ICON_DEFAULT = exports.PREFIXES = exports.PREFIX_DEFAULT = exports.LEDGER_CONN_DEFAULT = exports.LEDGER_CONN = exports.ENDPOINTS = exports.ENDPOINT_DEFAULT = exports.CRYPTOS_LEDGER = exports.CRYPTOS_ETH = exports.CRYPTOS = void 0;
var crypto_js_1 = require("./crypto.js");
Object.defineProperty(exports, "CRYPTOS", { enumerable: true, get: function () { return crypto_js_1.CRYPTOS; } });
Object.defineProperty(exports, "CRYPTOS_ETH", { enumerable: true, get: function () { return crypto_js_1.CRYPTOS_ETH; } });
Object.defineProperty(exports, "CRYPTOS_LEDGER", { enumerable: true, get: function () { return crypto_js_1.CRYPTOS_LEDGER; } });
var endpoints_js_1 = require("./endpoints.js");
Object.defineProperty(exports, "ENDPOINT_DEFAULT", { enumerable: true, get: function () { return endpoints_js_1.ENDPOINT_DEFAULT; } });
Object.defineProperty(exports, "ENDPOINTS", { enumerable: true, get: function () { return endpoints_js_1.ENDPOINTS; } });
var ledger_js_1 = require("./ledger.js");
Object.defineProperty(exports, "LEDGER_CONN", { enumerable: true, get: function () { return ledger_js_1.LEDGER_CONN; } });
Object.defineProperty(exports, "LEDGER_CONN_DEFAULT", { enumerable: true, get: function () { return ledger_js_1.LEDGER_CONN_DEFAULT; } });
var ss58_js_1 = require("./ss58.js");
Object.defineProperty(exports, "PREFIX_DEFAULT", { enumerable: true, get: function () { return ss58_js_1.PREFIX_DEFAULT; } });
Object.defineProperty(exports, "PREFIXES", { enumerable: true, get: function () { return ss58_js_1.PREFIXES; } });
var ui_js_1 = require("./ui.js");
Object.defineProperty(exports, "ICON_DEFAULT", { enumerable: true, get: function () { return ui_js_1.ICON_DEFAULT; } });
Object.defineProperty(exports, "ICON_DEFAULT_HOST", { enumerable: true, get: function () { return ui_js_1.ICON_DEFAULT_HOST; } });
Object.defineProperty(exports, "ICONS", { enumerable: true, get: function () { return ui_js_1.ICONS; } });
Object.defineProperty(exports, "NOTIFICATION_DEFAULT", { enumerable: true, get: function () { return ui_js_1.NOTIFICATION_DEFAULT; } });
Object.defineProperty(exports, "UIMODE_DEFAULT", { enumerable: true, get: function () { return ui_js_1.UIMODE_DEFAULT; } });
Object.defineProperty(exports, "UIMODES", { enumerable: true, get: function () { return ui_js_1.UIMODES; } });
Object.defineProperty(exports, "UITHEME_DEFAULT", { enumerable: true, get: function () { return ui_js_1.UITHEME_DEFAULT; } });
Object.defineProperty(exports, "UITHEMES", { enumerable: true, get: function () { return ui_js_1.UITHEMES; } });
exports.CAMERA_DEFAULT = 'off';
exports.CAMERA = [
    {
        info: 'on',
        text: 'Allow camera access',
        value: 'on'
    },
    {
        info: 'off',
        text: 'Do not allow camera access',
        value: 'off'
    }
];
exports.LANGUAGE_DEFAULT = 'default';
exports.LOCKING_DEFAULT = 'session';
exports.LOCKING = [
    {
        info: 'session',
        text: 'Once per session',
        value: 'session'
    },
    {
        info: 'tx',
        text: 'On each transaction',
        value: 'tx'
    }
];
exports.METADATA_UP_DEFAULT = 'off';
exports.METADATA_UP = [
    {
        info: 'off',
        text: 'Do not auto-update extension metadata',
        value: 'off'
    },
    {
        info: 'on',
        text: 'Auto-update extension metadata',
        value: 'on'
    }
];
exports.STORAGE_DEFAULT = 'off';
exports.STORAGE = [
    {
        info: 'on',
        text: 'Allow local in-browser account storage',
        value: 'on'
    },
    {
        info: 'off',
        text: 'Do not allow local in-browser account storage',
        value: 'off'
    }
];
