// @polkadot/ui-settings/cjs/defaults/ui.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NOTIFICATION_DEFAULT = exports.ICONS = exports.ICON_DEFAULT_HOST = exports.ICON_DEFAULT = exports.UITHEMES = exports.UITHEME_DEFAULT = exports.UIMODES = exports.UIMODE_DEFAULT = exports.LANGUAGE_DEFAULT = void 0;
const type_js_1 = require("./type.js");
exports.LANGUAGE_DEFAULT = 'default';
exports.UIMODE_DEFAULT = !type_js_1.isPolkadot && typeof window !== 'undefined' && window.location.host.includes('ui-light')
    ? 'light'
    : 'full';
exports.UIMODES = [
    {
        info: 'full',
        text: 'Fully featured',
        value: 'full'
    },
    {
        info: 'light',
        text: 'Basic features only',
        value: 'light'
    }
];
exports.UITHEME_DEFAULT = type_js_1.isPolkadot
    ? 'polkadot'
    : 'substrate';
exports.UITHEMES = [
    {
        info: 'polkadot',
        text: 'Polkadot',
        value: 'polkadot'
    },
    {
        info: 'substrate',
        text: 'Substrate',
        value: 'substrate'
    }
];
exports.ICON_DEFAULT = 'default';
exports.ICON_DEFAULT_HOST = type_js_1.isPolkadot
    ? 'polkadot'
    : 'substrate';
exports.ICONS = [
    {
        info: 'default',
        text: 'Default for the connected node',
        value: 'default'
    },
    {
        info: 'polkadot',
        text: 'Polkadot',
        value: 'polkadot'
    },
    {
        info: 'substrate',
        text: 'Substrate',
        value: 'substrate'
    },
    {
        info: 'beachball',
        text: 'Beachball',
        value: 'beachball'
    }
];
exports.NOTIFICATION_DEFAULT = 'popup';
